cmd/router/main.go

```diff
@@ -38,8 +38,13 @@ import (
 
 var log = logf.Log.WithName("InferenceGraphRouter")
 
-func callService(serviceUrl string, input []byte) ([]byte, error) {
-	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
+func callService(serviceUrl string, input []byte, headers http.Header) ([]byte, error) {
+	client := http.Client{}
```

Comment:
Should this http client be cached in some way?  we probably need to optimize here to make sure connections are reused by url.

Reply:
@yuzisun : Are you talking about reusing underlying TCP connections for making further HTTP requests ?
If that is the concern then I did a quick test and observed that when we are using 
`body, err := ioutil.ReadAll(resp.Body)` at line [53](https://github.com/kserve/kserve/pull/2396/files#diff-d8ccd6fde43fcb90d60f6319f818662f9e4340295fc9bbde755fd93d98da33c1R53) then multiple TCP connections are not being opened and same TCP connection is being used for new http requests. See image below:
![connection_reused](https://user-images.githubusercontent.com/8099140/187487167-1075e166-0eab-4667-a058-b1430af7341a.png)


Where as when I am commenting out line 53 then I am seeing multiple TCP connections being put into TIME_WAIT state. See attached image:
![connection_not_reused](https://user-images.githubusercontent.com/8099140/187486682-3ab18e84-2186-4ad3-83a7-aa435c720e3f.png)

So, if that is the concern then I guess we are good. You can check my gist here that I used to test this : https://gist.github.com/rachitchauhan43/8b7fc6a24074d0d0c0d9095f4e583a07

Also, referred official go docs to verify that we are correctly using `http` client from package.
See here: https://go.dev/src/net/http/response.go#L59

Reply:
Thanks for the detailed answer, I am actually referring to reusing the http connection across different requests, currently we are creating the client for each request. From the net http doc

Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used.

Reply:
Ok. Will take care of that.

Reply:
@yuzisun : Since, I am not changing any config of client at this time and only dealing with updating `Request` object so reverted back to using global http client provided by net/http package.